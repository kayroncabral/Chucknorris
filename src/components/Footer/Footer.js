  import React from 'react';
import styles from './Footer.scss'
import LinkedInIcon from '../../styles/icons/linkedin-icon.svg'
import GithubIcon from '../../styles/icons/github-icon.svg'

const Footer = () => {

  const socialIcons = [
    {
      name: 'Github',
      url: 'https://github.com/kayroncabral/Chucknorris',
      icon: <GithubIcon/>
    },
    {
      name: 'LinkedIn',
      url: 'https://www.linkedin.com/in/kayroncabral/',
      icon: <LinkedInIcon/>
    },
  ]

  const renderSocialIcon = (socialShortcut, index) => {
    return (
      <a className={styles.link} key={index} href={socialShortcut.url}>{socialShortcut.icon}</a>
    )
  }

  return (
    <div className={styles.Footer}>
      <div className={styles.content}>
        <div className={styles.socialShortcuts}>
          {
            socialIcons.map(renderSocialIcon)
          }
        </div>
        <div className={styles.rightsReserved}><p>KAYRON CABRAL Â© 2018. ALL RIGHTS RESERVED.</p></div>
      </div>
    </div>
  )
}

export default Footer
