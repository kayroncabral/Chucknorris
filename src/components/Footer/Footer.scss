@import '../../styles/styles.scss';

.Footer {
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: white;
}

.links {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  > .link {
    margin: $m-size;

    &:hover {
      border-bottom: none;
    }
  }
}

.socialShortcuts {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  > .link {
    margin: $s-size;

    &:hover {
      border-bottom: none;
    }
  }
}

.rightsReserved {
  display: flex;
  justify-content: center;
  text-align: center;
  padding: 15px;

  > p {
    margin: 0;
  	font-size: $s-size;
  	line-height: 1.5;
  	text-align: center;
  	color: $quick-silver;
  }
}
