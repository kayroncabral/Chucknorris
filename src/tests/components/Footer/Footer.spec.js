import React from 'react'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import Footer from '../../../components/Footer'

test('should render Header correctly', () => {
  const renderer = new ReactShallowRenderer()

  renderer.render(<Footer/>)
  expect(renderer.getRenderOutput()).toMatchSnapshot()
})
