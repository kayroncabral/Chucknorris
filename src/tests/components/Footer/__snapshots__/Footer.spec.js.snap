// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<footer
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <a
        className={[Function]}
        href="https://github.com/kayroncabral/Chucknorris"
      >
        <github-icon.svg />
      </a>
      <a
        className={[Function]}
        href="https://www.linkedin.com/in/kayroncabral/"
      >
        <linkedin-icon.svg />
      </a>
    </div>
    <div
      className={undefined}
    >
      <p>
        KAYRON CABRAL Â© 2018. ALL RIGHTS RESERVED.
      </p>
    </div>
  </div>
</footer>
`;
